R/colourInput.R

```diff
@@ -221,10 +221,16 @@ formatHEXsingle <- function(x) {
   if (x %in% grDevices::colors()) {
     x <- do.call(grDevices::rgb, as.list(grDevices::col2rgb(x) / 255))
   }
-  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6})$", x)) {
+  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6}|[[:xdigit:]]{8})$", x)) {
     stop(sprintf("%s is not a valid colour", x), call. = FALSE)
   }
 
+  # check whether it is 6-digit hex code with alpha.
+  if (substr(x, 1, 1) == "#" && nchar(x) == 9) {
```

Comment:
You're right, R doesn't support the [shorthand notation of hex colours](https://en.wikipedia.org/wiki/Web_colors#Shorthand_hexadecimal_form) (3 digits). But as someone who works a lot with colours (mostly on the web), I wanted to support that because I find it very convenient to be able to specify 3-digit colour codes. You'll often find in HTML code (or in CSS to be exact) that people use #0f0 instead of #00ff00. And the majority of the code for colourpicker is actually implemented in the javascript library I built for this, so that it could be used for HTML and not just for R.  

To answer my own question: it does look like [4-digit specification is indeed valid](https://drafts.csswg.org/css-color/#hex-notation)  (again, not in R, but in CSS, and I want to closely mimic web colours)

Does that make sense?